### PR TITLE
Gateway 3050 - Part III

### PR DESCRIPTION
**Pull request by mlandis5am**
This adds configuration for edge clients on the MessageHandler, more tests  and cleanup.

### CODE REVIEW

File: Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java [hunk 1]
```diff
@@ -155,7 +155,10 @@ public void sendMdn(Address recipient, MessageProcessResult result) {
     public int handleMessages(MessageHandler handler) {
 
         Session session = getMailSession();
-
+        session.setDebug(true);
+        session.setDebugOut(System.out);
```
Comment (msweaver) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java:159]: should these be here?

Comment (mlandis5am) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java:159]: those need to come out.

On Thu, Nov 15, 2012 at 11:42 AM, msweaver notifications@github.com wrote:

> In
> Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java:
> 
> > @@ -155,7 +155,10 @@ public void sendMdn(Address recipient, MessageProcessResult result) {
> >      public int handleMessages(MessageHandler handler) {
> > 
> > ```
> >      Session session = getMailSession();
> > ```
> > 
> > ## 
> > -        session.setDebug(true);
> > -        session.setDebugOut(System.out);
> 
> should these be here?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/CONNECT-Solution/CONNECT/pull/174/files#r2140962.

## 

Michael Landis
Software Engineer
www.5amsolutions.com | Software for Life(TM)
phone: 866.526.6042 x774
email: mlandis@5amsolutions.com
skype: mlandis5am

Comment (mlandis5am) [inline Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java:159]: .... or be configurable ...

On Thu, Nov 15, 2012 at 11:43 AM, Michael Landis
mlandis@5amsolutions.comwrote:

> those need to come out.
> 
> On Thu, Nov 15, 2012 at 11:42 AM, msweaver notifications@github.comwrote:
> 
> > In
> > Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/DirectMailClient.java:
> > 
> > > @@ -155,7 +155,10 @@ public void sendMdn(Address recipient, MessageProcessResult result) {
> > >      public int handleMessages(MessageHandler handler) {
> > > 
> > > ```
> > >      Session session = getMailSession();
> > > ```
> > > 
> > > ## 
> > > -        session.setDebug(true);
> > > -        session.setDebugOut(System.out);
> > 
> > should these be here?
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/CONNECT-Solution/CONNECT/pull/174/files#r2140962.
> 
> ## 
> 
> Michael Landis
> Software Engineer
> www.5amsolutions.com | Software for Life(TM)
> phone: 866.526.6042 x774
> email: mlandis@5amsolutions.com
> skype: mlandis5am

## 

Michael Landis
Software Engineer
www.5amsolutions.com | Software for Life(TM)
phone: 866.526.6042 x774
email: mlandis@5amsolutions.com
skype: mlandis5am

File: Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java [hunk 2]
```diff
@@ -49,12 +52,21 @@
...
+
+    @Autowired
+    private MessageHandler inboundMessageHandlerSoap;
+
     /**
      * Set up keystore for test.
      */
     @BeforeClass
     public static void setUpClass() {
-        DirectUnitTestUtil.writeSmtpAgentConfig();
```
Comment (msweaver) [inline Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java:57]: shouldn't this use the one from the util?

File: Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java [hunk 3]
```diff
@@ -81,11 +93,33 @@ public void canDistinguishInternalExternal() {
...
+        assertNotSame(outboundMessageHandler, inboundMessageHandlerSoap);
+        assertNotSame(inboundMessageHandlerSoap, inboundMessageHandlerSmtp);
+    }    
+
     /**
      * Tear down keystore created in setup.
      */
     @AfterClass
     public static void tearDownClass() {
-        DirectUnitTestUtil.removeSmtpAgentConfig();
```
Comment (msweaver) [inline Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java:89]: shouldn't this use the one from the util?

Comment (mlandis5am) [inline Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java:89]: i just statically imported them - they are being used from the util class.
 funnily enough i _think_ we both made that change in parallel.

On Thu, Nov 15, 2012 at 11:47 AM, msweaver notifications@github.com wrote:

> In
> Product/Production/Services/DirectCore/src/test/java/gov/hhs/fha/nhinc/direct/DirectMailClientSpringTest.java:
> 
> > ```
> >  /**
> >   * Tear down keystore created in setup.
> >   */
> >  @AfterClass
> >  public static void tearDownClass() {
> > ```
> > -        DirectUnitTestUtil.removeSmtpAgentConfig();
> 
> shouldn't this use the one from the util?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/CONNECT-Solution/CONNECT/pull/174/files#r2141037.

## 

Michael Landis
Software Engineer
www.5amsolutions.com | Software for Life(TM)
phone: 866.526.6042 x774
email: mlandis@5amsolutions.com
skype: mlandis5am

**Pull request merged**